packages/amplify_api/example/ios/unit_tests/GraphQLApiUnitTests.swift

```diff
@@ -226,9 +228,151 @@ class GraphQLApiUnitTests: XCTestCase {
+        
+        class MockStreamHandler: GraphQLSubscriptionsStreamHandler {
+            override func sendEvent(payload: [String : Any?]?, id: String, type: GraphQLSubscriptionEventTypes) {
+                            eventSentExp?.fulfill()
+                            let data = payload?["data"] as! String
+                            XCTAssertEqual("result data", data)
+                            XCTAssertEqual(GraphQLSubscriptionEventTypes.DATA, type)
+                            XCTAssertEqual("someCode", id)
+                }
+                }
+        
+        
+        FlutterGraphQLApi.subscribe(
+            flutterResult: { (result) -> Void in
+                XCTFail()
+            },
+            request: testRequest,
+            bridge: MockApiBridge(),
+            graphQLSubscriptionsStreamHandler: MockStreamHandler()
+        )
+        wait(for: [eventSentExp!], timeout: 1)
+    }
+    
+    func test_subscribe_api_error() throws {
+        let testRequest: [String: Any] = [
+            "document": "test document",
+            "variables": ["test key":"test value"],
+            "cancelToken" : "someCode"
+        ]
+        
+        class MockApiBridge: ApiBridge {
+            override func subscribe<ResultType>(request: GraphQLRequest<ResultType>,
+                                       valueListener: GraphQLSubscriptionOperation<ResultType>.InProcessListener?,
+                                       completionListener: GraphQLSubscriptionOperation<ResultType>.ResultListener?)
+                  -> GraphQLSubscriptionOperation<ResultType> {
+                completionListener?(.failure(APIError.invalidConfiguration("test error", "test recovery suggestion")))
+                let graphQLSubscriptionOperation = getMockGraphQLSubscriptionOperation(request: request)
+                return graphQLSubscriptionOperation
+            }
+        }
+        
+        class MockStreamHandler: GraphQLSubscriptionsStreamHandler {
+            override func sendEvent(payload: [String : Any?]?, id: String, type: GraphQLSubscriptionEventTypes) {
+                XCTFail()
+            }
+            
+            override func sendError(errorCode: String, details: [String: Any]) {
+                XCTFail()
+            }
+        }
+        
+        
+        FlutterGraphQLApi.subscribe(
+            flutterResult: { (result) -> Void in
+                if let exception = result as? FlutterError {
+                    XCTAssertEqual("AmplifyException", exception.code)
+                    XCTAssertEqual(FlutterApiErrorMessage.ESTABLISH_SUBSCRIPTION_FAILED.rawValue, exception.message)
+                    let errorMap: [String: Any] = exception.details as! [String : Any]
+                    XCTAssertEqual(
+                        ["platform": "iOS", "localizedErrorMessage": "test error", "recoverySuggestion": "test recovery suggestion"],
+                        errorMap["PLATFORM_EXCEPTIONS"] as? [String: String])
```

Comment:
Isn't this wrong?

Reply:
The tests were correct as per the original error handling approach. I've updated them now after the refactor.